### ISSUE TITLE
Current Software Install List for Raspberry Pi's

### ISSUE THREAD

**Issue by RachaelDinger** [CSUCyberSec/HPWebInterface#3]
1) Kippo - installed
2) Kippo Graph - installed
3) Snort - installed
4) Barnyard2 - installed
5) Apache2 - installed
6) MySQL - installed
7) PulledPork - installed
8) BASE - installed 
9) Ruby 1.9.3
10) RubyGems 2.4.6
11) Rails 4.2.0
12) bundler-1.8.4 (gem)
13) pdfkit 0.6.2 (gem)

Dependencies Installed:
libcap-1.6.2
libdnet-1.12
daq-2.0.4
apache2-utils
apache2.2-bin
apache2.2-common
libapache2-mod-php5
mysql-client
mysql-common
mysql-server
g++
libtool
flex 
bison 
gcc 
libnet1 
libnet1-dev 
libapache2-mod-php5 
libcrypt-ssleay-perl 
libpcre3 
libpcre3-dev 
libmysqlclient-dev 
libphp-adodb 
libssl-dev 
libwww-perl 
libmysqlclient-dev 
ntp 
php5-cli 
php5-gd 
php5-mysql 
php-pear
libxslt1-dev
libjpeg62-dev
ttf-dejavu
libtiff4-dev
libjasper-dev
libfontconfig1-dev
libopenexr-dev
libwmf-dev
librsvg2-dev
libfftw3-dev
liblzma-dev
liblcmsl-dev
graphviz-dev
libdjvulibre-dev
xorg
ImageMagick 6.9.0-10
wkhtmltopdf 0.12.2

**Comment by pyroesque** [comment]
sorry, should have put my comments for authy here, see issue 5.

**Comment by miles00** [comment]
I wasn't able to get Base working even after copied all the original files. It looks like the apache tries and fails to connect to 127.0.1.1 for some reason. Rachael, looks like you got farther than me with regards to this. I can not work on this more till Thursday after class because of 2 mid terms.

**Comment by RachaelDinger** [comment]
ok thanks Miles. It should be trying to connect to 127.0.0.1 though. I will work on it some more tomorrow.

**Comment by RachaelDinger** [comment]
Or I can just try to install a different mySQL workbench, as BASE really only provides access to the snort tables, which leaves us having to install something else to view the kippo database, or any other tables we may make to store info. I will probably completely uninstall BASE then --- because it really isn't working and from what others are saying, it seems pretty limited on what you can do with it.

**Comment by miles00** [comment]
The guide we've been looking at recommended Splunk but I've never heard of it. Maybe we should look into it? Yeah I'm not sure why Apache is trying to connect to 127.0.1.1 that's another random problem we go to look at.

**Comment by RachaelDinger** [comment]
I'm having a really hard time trying to get Snorby to install. I keep hitting an issue with dm-rails/railtie not being where the snorby rake file wants it to be. I thought it was a ruby install issue, so I've reinstalled and it's not making a difference. If anyone can provide some insight that would be great. I've googled this problem and not really gotten anywhere. Plus I've also noticed that snorby's website has been down? Anyone else notice this too?

**Comment by miles00** [comment]
I'll look into it more and let you know what I find. That's too bad though everyone was saying good things about Snorby.

**Comment by RachaelDinger** [comment]
Chris, to be more specific, I have snorby unpacked under /var/www/snorby. The ruby path, if you type "which ruby" is installed under /usr/local/bin. I run into a problem when the rake file tries to find the dm-rails, because it's looking for some weird path, I think it is set by the line: require File.path_extend(..boot, __FILE__). Then the error occurs at require 'dm-rails/railtie'. To me it looks like it's looking for a folder to levels up called boot.. But why would it? None of the tutorials I have seen address this

**Comment by pyroesque** [comment]
first make sure dm-rails is installed  [*gem list --local | grep dm-rails*] if it doesn't show up then install it with [*gem install dm-rails*]. if it is installed try changing the rakefile to point to the installation path, should be something like <ruby install path>/gems/ruby2.2.0/gems. Other then that I don't have many ideas. I can look into it more once I have my pi and can attempt an install, hopefully by the end of the week.

**Comment by RachaelDinger** [comment]
Hey, sorry my last comment should have been addressed to Miles, but somehow I thought that I was talking to Chris (this is what happens when you try to do things when you are brain dead). Thank you Chris for looking into it anyway. I did check to see that that dm-rails was installed, I also did a "bundle update" afterwards like all the help forums say to do -- the problem still persists. I really think it is a path issue that is set in the application.rb file that gets called when you run the rake command for snorby. I tried playing around with the variable for the path field yesterday, but I do not think I am getting it right. I will try to work on it some more today and post how it goes.

**Comment by miles00** [comment]
I'll let you know what I find too sometime around 2 I think.

**Comment by miles00** [comment]
I think I may have it installed, but I'm working on configuring it now. I'll let you know if it actually works.

**Comment by miles00** [comment]
Yep I'm getting the same error as you Rachael, and I'm not sure what's wrong because the paths for everything seem find. Let me know if you figure it out.

**Comment by miles00** [comment]
Hey guys, sorry all the posts on this issue. Rachael try this: 
bundle exec rake snorby:setup -I /var/lib/gems/1.9.1/gems/dm-rails-1.2.1/lib
I had to manually include dm-rails for some reason. It is still complaining about my config file but I got past that issue that you had.

**Comment by miles00** [comment]
Well, with the given library path I posted earlier it still cannot load railtie. The path looks 100% correct. I'm not sure of what else can be done at this point. I'm going to give up on this for now but Chris maybe you can figure it out.

**Comment by RachaelDinger** [comment]
Thanks Miles for looking. I'm going to uninstall the current version of snorby and re-install by following the directions on snorby's website tomorrow (the website is back up)

**Comment by RachaelDinger** [comment]
One thing I have noticed -- that is a discrepancy -- is that I installed a higher version of Image Magick than what snorby's website tells you to -- this could be part of the problem. So it looks like I will be fixing that tomorrow as well

**Comment by RachaelDinger** [comment]
Here is the latest of where we are at with the Snorby install. Yesterday, Miles tried to use .rvm to package install ruby and all of its dependencies -- this installation include the stable addition of ruby and rails, as well as the latest patches. I was incorrect in my assumption in the last comment that we had to use the older versions of the software, as the oldest listed is the minimal requirement only. Therefore we can rule out the software version being newer as a problem. However, Miles ran into the same issue with the rake file, so we are back at square one with this. 

Last night, after some digging, I did find a blog about installing snorby by a guy who boasts that he has found a way around these rake issues. I sent the blog to Miles. The link is here: https://n40lab.wordpress.com/2013/06/02/snorby-in-centos-6-4/ 
in step 11 he references having issues with bundling ruby 1.9.3 and proceeds to edit the gem file (which has a strong correlation with the gem file issues we are having). Either Miles or I will attempt this next to see if we can get around this issue. If this doesn't work, we can attempt to email the creators of Snorby -- I am not sure how far this is going to get us, however; as I have seen multiple posts for this problem with absolutely no answers given (which is seriously concerning). 
In the meantime, as Plan B, I have emailed Dr. Collins requesting a license for Aanval, which is another really nice front end for Snort. Licenses are not free, but you can download and install for a trial period. The other front ends for Snort are not as nice, but still get the job done. 1) Sguil is a valid alternative, and its free. 2) Squert -- also free, but its a very watered down alternative when compared to the others, and 3) Placid -- which is a text based GUI (personally I don't like this one at all). So those are the alternatives. If we cannot move forward with Snorby we need to decide on another plan of action and get to it -- as we are running out of time.

**Comment by RachaelDinger** [comment]
Today we were still unable to install Snorby. We followed the directions in step 11 of the guide -- it doesn't seem to help at all. I also noticed that Snorby's website was down (again). So I think it is safe to say that we are going to be looking into another front end. I still haven't heard back about the license for Aanval -- so I'm going to start looking into installing Sguil, as this seems to be the next alternative. 

On a positive note, Jared was able to get Kippo to start writing to the database, and Kippo-Graph began filling out "super cool" charts --- this was done on the Linux VM, so he will just have to reproduce what he did on the Pi -- should be a quick step.
